### PR TITLE
[Op][Internal] Reference implementation for internal::RMS op

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 - Reference implementation and ref tests for [internal::RMS](https://github.com/openvinotoolkit/openvino/blob/70142121c1c632b6d710026150b0e6deb20b0e68/src/common/transformations/include/ov_ops/rms.hpp) (moved from gpu custom ops) available for common transformations
 - Enable Accuracy tests for RMSFusion transformation
 
 The changes moving RMS from gpu to internal are from PR:
 -  #24539 (to be merged first)
 
 Internal RMS op docs:
- https://github.com/openvinotoolkit/openvino/pull/24564

 ### Tickets:
 - Related to 136262

### CODE REVIEW

File: src/core/reference/include/openvino/reference/rms_norm.hpp [hunk 1]
```diff
@@ -72,5 +73,33 @@ void rms_norm(const T* in,
...
+void rms_norm_mul_convert_out(const T_IN* in,
+                              const AxisSet& axes,
+                              T_OUT* out,
+                              const Shape& in_shape,
+                              double eps,
+                              const Shape& scale_shape,
+                              const T_IN* scale) {
+    std::vector<T_IN> tmp_out(shape_size(in_shape));
+    rms_norm(in, axes, tmp_out.data(), in_shape, eps, scale_shape, scale);
+    convert(tmp_out.data(), out, tmp_out.size());
```
Comment (praasz) [inline src/core/reference/include/openvino/reference/rms_norm.hpp:101]:
```suggestion
    rms_norm(in, axes, out, in_shape, eps, scale_shape, scale);
    convert(out, out, tmp_out.size());
```
The `out_temp` should not be required.

File: src/plugins/template/backend/ops/rms_internal.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    case element::bf16:
+        return evaluate<element::bf16>(as_type_ptr<op::internal::RMS>(node), outputs, inputs);
+    case element::f16:
+        return evaluate<element::f16>(as_type_ptr<op::internal::RMS>(node), outputs, inputs);
+    case element::f64:
+        return evaluate<element::f64>(as_type_ptr<op::internal::RMS>(node), outputs, inputs);
+    case element::f32:
+        return evaluate<element::f32>(as_type_ptr<op::internal::RMS>(node), outputs, inputs);
+    default:
+        OPENVINO_THROW("Unhandled data type ", node->get_element_type().get_type_name(), " in evaluate_node()");
```
Comment (praasz) [inline src/plugins/template/backend/ops/rms_internal.cpp:67]:
```suggestion
        OPENVINO_THROW("Unhandled data type ", node->get_input_element_type(0), " in evaluate_node()");
```
Use same type as for switch or make.

File: src/plugins/template/backend/ops/rms_internal.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+              ov::TensorVector& outputs,
+              const ov::TensorVector& inputs) {
+    using ET = typename ov::element_type_traits<T>::value_type;
+    const auto normalized_axes =
+        ov::util::normalize_axes(node->get_friendly_name(), std::vector<int64_t>{-1}, inputs[0].get_shape().size());
+
+    outputs[0].set_shape(inputs[0].get_shape());
+
+    const auto in_type = inputs[0].get_element_type();
+    const auto out_type = outputs[0].get_element_type();
```
Comment (praasz) [inline src/plugins/template/backend/ops/rms_internal.cpp:28]:
```suggestion
    const auto& in_type = inputs[0].get_element_type();
    const auto& out_type = outputs[0].get_element_type();
```

File: src/core/reference/include/openvino/reference/rms_norm.hpp (hunk 1 above)
Comment (mitruska) [reply src/core/reference/include/openvino/reference/rms_norm.hpp:101]: Could you please elaborate a little bit more? 

The goal is to convert f32 to f16 (when input data is f32 and output f16), the proposed change `convert(out, out, ...);` seems to have no effect.
The conversion should be optimized in plugins for sure, but here the other reference functions like `power` or `reduce_mean` inside the `rms_norm` require the in/out ptr to have the same type. 
So seems like the temp vector will be needed at some point of the ref impl anyway.

File: src/plugins/template/backend/ops/rms_internal.cpp (hunk 2 above)
Comment (mitruska) [reply src/plugins/template/backend/ops/rms_internal.cpp]: Done

File: src/plugins/template/backend/ops/rms_internal.cpp (hunk 3 above)
Comment (mitruska) [reply src/plugins/template/backend/ops/rms_internal.cpp]: Done

File: src/core/reference/include/openvino/reference/rms_norm.hpp (hunk 1 above)
Comment (praasz) [reply src/core/reference/include/openvino/reference/rms_norm.hpp:101]: The suggestion was wrong. But if `T_IN and T_OUT` are same type the additional buffer and convert (dummy copy) are not required.
The template could have special version for such case.

File: src/plugins/template/backend/ops/rms_internal.cpp (hunk 2 above)
Comment (praasz) [reply src/plugins/template/backend/ops/rms_internal.cpp]: There is no change at this line.

Comment (mitruska) [reply src/plugins/template/backend/ops/rms_internal.cpp]: Yes there is, it was commited within https://github.com/openvinotoolkit/openvino/pull/24556/commits/0f3b63cf2780c0f9263affd3ac5165ef4d3fdfaf

File: src/core/reference/include/openvino/reference/rms_norm.hpp (hunk 1 above)
Comment (mitruska) [reply src/core/reference/include/openvino/reference/rms_norm.hpp:101]: The `rms_norm` can be used if conversion is not needed. It expects `T` in/out of the same type.

https://github.com/openvinotoolkit/openvino/blob/10bc58263c756026690c843b32e5dbf1a255621f/src/core/reference/include/openvino/reference/rms_norm.hpp#L32-L33

https://github.com/openvinotoolkit/openvino/blob/10bc58263c756026690c843b32e5dbf1a255621f/src/core/reference/include/openvino/reference/rms_norm.hpp#L65-L72

**Review by praasz — APPROVED** [review]

**Review by mmikolajcz — APPROVED** [review]

**Pull request merged**